File: Servidor_java/Modelos/PowerBank.java
package Servidor_java.Modelos;

public class PowerBank extends Produto {
    
    private String marca;
    private String modelo;
    private int capacidade;

    public PowerBank(){
        super();
    }

    public PowerBank(int id,String nome, String descricao,double preco,int estoque,String marca,String modelo,int capacidade){

        super(id,nome,descricao,preco,estoque);

        this.marca = marca;
        this.modelo = modelo;
        this.capacidade = capacidade;

    }

    public String getMarca(){return marca;}
    public String getModelo(){return modelo;}
    public int getcapacidade(){return capacidade;} 

    public void setMarca(String marca){this.marca = marca;}
    public void setModelo(String modelo){this.modelo = modelo;}
    public void setcapacidade(int capacidade){this.capacidade = capacidade;}

    public String toString() {
        return "PowerBank [id=" + getId() + ", nome=" + getNome() + ", preco=" + getPreco() + ", estoque=" + getEstoque() + "Marca="+ marca + "modelo="+ modelo + "Duracao" + capacidade +"MA/h ]";
    }    

}

File: Cliente_cpp/src/modelos/PowerBank.cpp
#include "modelos/PowerBank.h"
#include <string>
#include <iostream>

PowerBank::PowerBank(int id, const std::string& nome, const std::string& descricao, double preco, int quantidade_estoque, const std::string& marca, const std::string& modelo, int capacidade)
                        : Produto(id, nome, descricao, preco, quantidade_estoque), marca(marca), modelo(modelo), capacidade(capacidade) {}

TipoProduto PowerBank::getTipo() const {
    return TipoProduto::POWERBANK;
}

int PowerBank::getCapacidade() const {
    return capacidade;
}

void PowerBank::setCapacidade(int capacidade) {
    this->capacidade = capacidade;
}

std::string PowerBank::getModelo() const {
    return modelo;
}

void PowerBank::setModelo(const std::string& modelo) {
    this->modelo = modelo;
}

std::string PowerBank::getMarca() const {
    return marca;
}

void PowerBank::setMarca(const std::string& marca) {
    this->marca = marca;
}

void PowerBank::mostrar() const {
    Produto::mostrar();
    std::cout << "Capacidade: " << capacidade << " mAh" << std::endl;
    std::cout << "Modelo: " << modelo << std::endl;
    std::cout << "Marca: " << marca << std::endl;
}